llvm/test/CodeGen/AArch64/selectiondag-order.ll

```diff
@@ -55,8 +55,9 @@ end:                                        ; preds = %body
 
 define i64 @simulateWithDbgDeclare(<2 x i32> %a) local_unnamed_addr  {
 entry:
+  %ptr = alloca i32
```

Comment:
Slightly awkward case here in that this _might_ degrade the test quality... the intent here seems to be to test that the debug declare does not cause any changes to the code in the CHECK set below, so making it no longer use `%rand` as an argument might interfere with that.

But otoh the test is non-specific; there's no description of how the debug declare might be expected to affect codegen, so I don't think I have a specific suggested change - you could add a `store i64 %rand, ptr %ptr`, or expand the check to cover the generated code for `%ptr`, but I also don't think it's a blocking issue if it stays as-is.

Reply:
Hm, adding a store like that would affect the existing CHECK lines, so I'm not sure that's better...

Reply:
Yeah, I'm not convinced there are any solid answers - the test is to check that "nothing bad happens" because of the presence of a dbg.declare, rather than checking for a specific defect, so it's not clear what properties would need to be maintained to keep the test valid. If there aren't any obvious solutions, I'm fine with the latest revision.